go-stash/client.go

```diff
@@ -1,11 +1,28 @@
-package http
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gostash
```

Comment:
Can we call this package `stash` to match the names of the other providers i.e. `github`/`gitlab` etc?

Reply:
Another package called `stash` will be created, that consumes the client declared here. This other package will implement the gitProvider interfaces.